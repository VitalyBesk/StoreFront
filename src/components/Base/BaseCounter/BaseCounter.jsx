import React from "react";
import PropTypes from "prop-types";
import "./BaseCounter.scss";

function BaseCounter({
  handleClickIncrement = () => {},
  handleClickDecrement = () => {},
  value = 0
}) {
  return (
    <div className="base-counter d-flex between-xs center-xs">
      <div className="base-counter__value d-flex  center-xs middle-xs">
        {value}
      </div>
      <div className="d-flex flex-column between-xs">
        <button onClick={handleClickIncrement} className="base-counter__btn">
          +
        </button>
        <button
          onClick={handleClickDecrement}
          disabled={!value}
          className="base-counter__btn"
        >
          -
        </button>
      </div>
    </div>
  );
}

BaseCounter.propTypes = {
  handleClickIncrement: PropTypes.func.isRequired,
  handleClickDecrement: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired
};

export default BaseCounter;
