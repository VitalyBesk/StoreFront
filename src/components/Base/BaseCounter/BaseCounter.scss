.base-counter {
  margin: 0.5rem;
  height: 3.25rem;
  width: 4.875rem;

  &__value {
    background: var(--light-gray-color-3);
    width: 3.25rem;
    height: 100%;
    color: var(--black-color);
  }

  &__btn {
    padding: 0.25rem;
    display: flex;
    align-items: center;
    justify-content: center;
    background: var(--light-gray-color-2);
    color: var(--black-color);
    border: none;
    width: 1.4375rem;
    height: 1.4375rem;
    margin-left: 0.25rem;
    cursor: pointer;
    outline: none;
    transition: background 0.3s ease;

    &:hover {
      background: var(--gray-color);
    }
  }
}
