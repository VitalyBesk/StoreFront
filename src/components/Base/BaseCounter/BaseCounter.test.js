import React from "react";
import BaseCounter from "./BaseCounter";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("<BaseCounter/>", () => {
  let count = 0;

  const mockFnIncrement = jest.fn(() => (count += 1));
  const mockFnDecrement = jest.fn(() => (count -= 1));

  const wrapper = mount(
    <BaseCounter
      handleClickIncrement={mockFnIncrement}
      handleClickDecrement={mockFnDecrement}
      value={count}
    />
  );

  it("snapshot render", () => {
    const component = renderer.create(
      <BaseCounter
        handleClickIncrement={mockFnIncrement}
        handleClickDecrement={mockFnDecrement}
        value={count}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("checking the type of props", () => {
    expect(typeof wrapper.props().value).toBe("number");
    expect(typeof wrapper.props().handleClickIncrement).toBe("function");
    expect(typeof wrapper.props().handleClickDecrement).toBe("function");
  });

  it("it should increment count when button +(plus) clicked", () => {
    expect(wrapper.props().value).toBe(0);
    wrapper.props().handleClickIncrement();
    expect(mockFnIncrement).toHaveBeenCalled();
    wrapper.setProps({ value: count });
    expect(wrapper.props().value).toBe(1);
  });

  it("it should increment count when button -(minus) clicked", () => {
    count = 1;
    expect(wrapper.props().value).toBe(1);
    wrapper.props().handleClickDecrement();
    expect(mockFnDecrement).toHaveBeenCalled();
    wrapper.setProps({ value: count });
    expect(wrapper.props().value).toBe(0);
  });

  it("it should set button -(minus) disabled when value is equal 0", () => {
    expect(wrapper.props().value).toBe(0);
    expect(wrapper.find("button:last-child").prop("disabled")).toBeTruthy();
  });
});
