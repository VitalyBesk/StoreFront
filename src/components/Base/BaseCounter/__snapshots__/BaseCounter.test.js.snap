// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseCounter/> snapshot render 1`] = `
<div
  className="base-counter d-flex between-xs center-xs"
>
  <div
    className="base-counter__value d-flex  center-xs middle-xs"
  >
    0
  </div>
  <div
    className="d-flex flex-column between-xs"
  >
    <button
      className="base-counter__btn"
      onClick={[MockFunction]}
    >
      +
    </button>
    <button
      className="base-counter__btn"
      disabled={true}
      onClick={[MockFunction]}
    >
      -
    </button>
  </div>
</div>
`;
